api/batch.go

```diff
@@ -81,3 +88,51 @@ func makeCategoriesBatchRoute(router gin.IRouter) {
 		ginutils.RenderSuccess(c, batch)
 	})
 }
+
+// @Description Get multiple addresses from naming service
+// @ID naming_service
+// @Summary Get list of addresses for passed coins
+// @Accept json
+// @Produce json
+// @Tags ns
+// @Param coins body api.ENSBatchRequest true "ns name and coins"
+// @Success 200 {object} api.LookupBatchPage
+// @Router /v2/ns/lookup [post]
+func makeNsLookupRoute(router gin.IRouter) {
+	router.POST("/ns/lookup", func(c *gin.Context) {
+		var req ENSBatchRequest
+		if err := c.BindJSON(&req); err != nil {
+			ginutils.ErrorResponse(c).Message(err.Error()).Render()
+			return
+		}
+
+		lookupsChan := make(chan blockatlas.Resolved)
```

Comment:
@dpereskokov can you open other PR to make the others batch request asynchronous? 😄

Reply:
This one is really time consuming. Checking one coin takes ~5 seconds

Reply:
I don't this is the best way to handle batch query, instead of calling `handleLookup` one by one, we should add `handleBatchLookup` (or something smilar) to `ens.go` and `zns.go`

For example, we can try batch JSON RPC request for ENS, and for UD, if you take a look at the response, we don't need to send multiple request to get all the coins